app/controllers/questionnaire_controller.rb

```diff
@@ -0,0 +1,53 @@
+class QuestionnaireController < ApplicationController
+	def index
+
+		# @display is an array of hashes, containing question and its answers 
+		# as Rows in Table Answers. something like, 
+		# [{:question => "What's my name?", :answer => AnswersTable},...]
+		@display = Array.new {Hash.new}
+		@selected_a_id = []
+		@end_of_questionnaire = false	#set default value
+		@response = ""
+
+		if params[:form] != nil
+			aid = params[:form]
+			if aid == "3"
+			 	@response = "Good choice"
+			elsif aid == "4"
+			 	@response = "Hmmmmm"
+			elsif aid == "5"
+			 	@response = "Ewwwwww"
+			elsif aid == "6"
+			 	@response = "I knew it!"
+			elsif aid == "7"
+			 	@response = "Cool"
+			elsif aid ==  "8"
+			 	@response = "MANGO HABANERO"
+			elsif aid == "9"
+			 	@response = "So you like to be second."
+			end
+		end 
+
+		current_qid = 1	#base case, at root question
+		if params[:ans] != nil	#got an answer
+			a = Answers.find(params[:ans])
+			if a.leads_to == nil	#case1: a is the last answer
+				@end_of_questionnaire = true
+				@selected_a_id.push(a.id) 	#corner case: add the last selected answer into array
+				current_qid = a.questionnaire_id
+			else	#case2: a is an intermidate answer
+				current_qid = a.leads_to
+			end
+		end
+
+		while current_qid != nil #populate @display with q/a pairs
+			q = Questionnaire.find(current_qid)
+			as = Answers.where(questionnaire_id: current_qid)
+			@display.unshift({:question => q.question, :answer => as})
+			if current_qid != 1  
```

Comment:
Since you use the 1 to compare to the root question, you could make that into a constant like Question.root_question_id, or something like that from the question model